main.cpp

```diff
@@ -56,15 +56,25 @@ Purpose:  This project will show you the difference between member functions and
 #include <string>
 struct T
 {
-    T(<#type name#> v, const char* <#variable name#>)   //1
-    //2
-    //3
+    T(double v, const char* namePtr) : value(v) 
+    {
+        if (namePtr != nullptr) name = *namePtr;
+        else name = "";
+    }   //1
+
+    double value;//2
+    std::string name;//3
 };
 
-struct <#structName1#>                                //4
+struct CompareClass                                //4
 {
-    <#type name#> compare(<#type name#> a, <#type name#> b) //5
+    T* compare(T* a, T* b) //5
     {
+        if (a == nullptr || b == nullptr) 
+        {
+            std::cout << "Invalid pointer passed to compare function." << std::endl;
+            return nullptr;
+        }
         if( a->value < b->value ) return a;
         if( a->value > b->value ) return b;
         return nullptr;
```

Comment:
D.R.Y.
you've now got 2 `return nullptr;` statements in this function
Restructure the logic so you only have 1 `return nullptr` statement.